src/org/opensolaris/opengrok/history/Repository.java

```diff
@@ -364,6 +364,8 @@ final void createCache(HistoryCache cache, String sinceRevision)
      */
     abstract String determineBranch() throws IOException;
 
+    abstract String determineCurrentVersion() throws IOException;
```

Comment:
I wonder why it's made abstract when you return `null` in all repositories except `GitRepository`. Also this is about the other `determineBranch` and `determineParent`. @tarzanek